Refinement, continued: black and white

garrett opened the pull request:
Continuation of #8.

Moved the commits of the black and white versions to this branch (and PR).

ifohancroft commented:
1. How do you squash multiple commits into a single one?
2. Regarding your idea to have a simplified and true to original versions of the black and white and black and white inverted Tux - I don't mind either way. To be honest, I liked your simplified version more but since I wanted to be able to include the changes from the colorful version and I suck at designing things, instead of simplifying the paths myself, I just cheated and used them as they are in the regular Tux.svg. Although, maybe it does make sense to have both versions.

garrett commented:
@ifohancroft: I've moved your changes to the black and white version to this pull request.

I've additionally re-added my simplified black and white versions.

As there are now so many different variants, I've moved your B&W version to `black-and-white` subdirectory and mine to `black-and-white-simplified`.

We'll add the sticker version to a `sticker` directory in an upcoming PR.

ifohancroft commented:
@garrett Sounds good! Tag me when you make the feet adjustments to the sticker version so I can confirm with Sticker Mule that it's ready. Btw did I upload the pdf from that shows the outline in the other PR in my comment or did I forget?

garrett commented:
> How do you squash multiple commits into a single one?

`git rebase` with various flags. It's dangerous though... you should only rebase locally, not on shared branches, except when you let the other person know or are about to commit a feature branch. (GitHub also has rebase and squashing available.)

Since I wanted to get your changes in ASAP with the main Tux and was going to merge the branch in anyway, I went ahead and did the rebase to squash things down (with a `git rebase -i master` to rebase interactively compared to the master branch).

> Regarding your idea to have a simplified and true to original versions of the black and white and black and white inverted Tux - I don't mind either way. To be honest, I liked your simplified version more but since I wanted to be able to include the changes from the colorful version and I suck at designing things, instead of simplifying the paths myself, I just cheated and used them as they are in the regular Tux.svg. Although, maybe it does make sense to have both versions.

I could see if we can take the best of each version. But that'll take time.

The simplified version does have a little bit of stylistic optimizations that I did and it's a combined single path instead of grouped shapes, which is useful in many circumstances. However, if someone wants to repose Tux — like the KVM logo does, for example — it's useful to have a grouped path version. (FWIW: I made a revised KVM logo, including a black and white version, many years ago. It was printed on a bunch of mugs among other things.)

So there's definitely a use for both, and we can optimze each.

I split this out from the other since this all takes time and we should communicate about it instead of delaying your (much appreciated!) fixes to the main `tux.svg`.

garrett commented:
> Btw did I upload the pdf from that shows the outline in the other PR in my comment or did I forget?

You did!

I should open an issue with the PDF so we don't lose it.

garrett commented:
OK, I've created the sitcker issue @ #10.

I'll try to get this done soon (this weekend).

garrett commented:
Sorry for the delays, BTW. Thanks for being patient and thanks for all your contributions!

ifohancroft commented:
> `git rebase` with various flags. It's dangerous though... you should only rebase locally, not on shared branches, except when you let the other person know or are about to commit a feature branch. (GitHub also has rebase and squashing available.)
> 
> Since I wanted to get your changes in ASAP with the main Tux and was going to merge the branch in anyway, I went ahead and did the rebase to squash things down (with a `git rebase -i master` to rebase interactively compared to the master branch).

Thank you! I have been learning git on per need basis, so there are still things that I need to learn.

> The simplified version does have a little bit of stylistic optimizations that I did and it's a combined single path instead of grouped shapes, which is useful in many circumstances. However, if someone wants to repose Tux — like the KVM logo does, for example — it's useful to have a grouped path version.

Would you please teach me that? I was wondering how you did it since even Path > Break Apart doesn't separate it. Again, yet another thing I did (leaving the paths separate and just grouping them) not because I thought it would be better but because I didn't know how to do it as a single path like you did :D

> (FWIW: I made a revised KVM logo, including a black and white version, many years ago. It was printed on a bunch of mugs among other things.)
> So there's definitely a use for both, and we can optimze each.

That's awesome!

> I split this out from the other since this all takes time and we should communicate about it instead of delaying your (much appreciated!) fixes to the main `tux.svg`.

Thank you!

This reminds me: I am not sure how image upload and update rights work in Wikipedia and I don't know if it was uploaded by you but perhaps we should the svg/png in there as well? I mean it's using your version.

No problem. Sorry for the 'bump' comments :D

Actually that's not a bad idea. Perhaps I should do it from time to time to let my mind get some free time :D

ifohancroft commented:
Also, let me know if there's something you'd like me to do to the black and white versions.

ifohancroft commented:
After we are done with the original version, perhaps we should get back to working on this one.

I am not sure whether we established what we are about to do here.

1. Are we going to create a single version that includes the original Tux improvements that also includes your stylistic changes?
2. Should we have it as a single path or should we keep the body parts separate but group them?

I'm thinking that ideally, we should be able to have the original Tux + the two black and white versions (inverted and not inverted) in a single branch, maybe even in a single folder (I mean in the root).

I don't know how you make the simplified version and the stylistic changes. Is it possible to turn the original design with the improvements black and white, then make your stylistic changes and simplification, then merge into a single path? If it's possible to have the merge along with the other steps and have it as a last step, perhaps we can leave it unmerged or have a separate layer that's the merged version (left visible and the grouped version layer disabled by default)
